Updated fix for issue whereby stream language is obscured by live banner

ross39 opened the pull request:
![Screenshot 2023-02-09 at 00 24 13](https://user-images.githubusercontent.com/20873258/217683422-1b561c96-aab2-46fb-a9c0-e61ea270cdc7.png)


Please see above screenshot for updated fix. It has been moved from below the subscribe button to just left of the live banner. Any feedback is appreciated.

ross39 commented:
I should add this is a fix for issue #12141

schlawg commented:
By the ribbon is problematic on mobile.

![IMG_835E73717CFC-1](https://user-images.githubusercontent.com/101470903/217943001-8f8e6017-29cb-4dc4-8063-887ac2c80ff8.jpeg)

benediktwerner commented:
Well, it will just move to the next line like it already does for the SLCC.

If we don't want that, next to the subscribe button is the only spot I guess. But it seems like a rather odd place for it.

schlawg commented:
You're right, beside the ribbon is actually a non-issue on portrait mobile.  It's viewport widths just _above_ the picture breakpoint (making height static) with long streamer names that have possible issues.  I suppose we could worry about those if/when they crop up.

If we go beside the ribbon then, it should be margin-right because the ribbon is always on the right but the language text positioning varies by text direction.  Too bad we can't just replace the "Live!" text with the audio language but I suppose that wouldn't be fair to new users who don't know what the blue ribbon means.

ross39 commented:
I added div as when I tried margin-right it was still being obscured by the Live tag when using the span tag. I understand div is a block element and span is used for inline element so span is to be preferred but the text wasn't moving at all except for when I changed it to div. I will revisit this as clearly I must be doing something wrong as margin-right should work. Display:inline is a bit ugly, so I will change that. I will add the `margin-#{$end-direction} ` for rtl languages. 



> FYI, if you add `Fixes #123` or `Closes #123` to the initial comment that issue will be auto-closed when the PR gets merged. Linked it manually now.
> 
> Also, is there a reason for the `span` -> `div` change? And also not sure I understand why `display: inline` is needed and probably wouldn't use `width` here. Also, it seems to now always push it over even if a streamer isn't live?
> 
> It seems like just adding some `margin-right` when the streamer is live should work. Probably needs a separate value on small screens where the ribbon is slimmer. And ig really should be `margin-#{$end-direction}` to consider rtl languages.

ross39 commented:
![Screenshot 2023-02-21 at 22 28 55](https://user-images.githubusercontent.com/20873258/220473811-5a29f495-60a7-4d9d-b00f-284591bbc8d9.png)


Hi, I changed to `margin-#{$end-direction}` and got rid of the div and reverted back to span. I include a screenshot of the results of the change for clarity. Any issues let me know.

schlawg commented:
Did you check it with Arabic to make sure the margin-#{$end-direction} change does what you think?

ross39 commented:
![Screenshot 2023-02-22 at 13 55 03](https://user-images.githubusercontent.com/20873258/220640647-c8b422ee-81c7-466f-b09f-3a7145fde625.png)


Changing `lastStreamLang = some("En")` to `lastStreamLang = some("Ar")` seems to still have the desired effect. I have included a screenshot so let me know what you think. I am happy to make changes if needed

ross39 commented:
the writing seems a bit small but I think Arabic fonts are normally smaller than English fonts?

schlawg commented:
I mean change **your** language to arabic to see if the margin-#{$end-direction} change does what you want it to.  Select an rtl language in your profile/account dropdown upper right.

I do not think margin-#{$end-direction} is what you want, but I want you to investigate and find out why if it's not.  It's a good habit to check these things yourself before a PR to alleviate some of the burden on maintainers.

schlawg commented:
P.S. I know Benedikt suggested the end-direction thing but he is also very busy and stretched very thin.  Even if thibault himself suggests a revision to someone's PR, ideally they would still verify it after making it.

ross39 commented:
Hi, apologies for the delay in getting back to this pr as I have been busy with life but I want to get this pr resolved and contribute more to lichess. I will ensure I verify all revision suggestions before resubmitting. I am finding this Rtl stuff kinda confusing. So what you're saying is that `margin#{$end-direction}`(I realize now in my haste to push the change I did not verify this and the code is actually incorrect(margin-#{$end-direction}), should instead be `margin-right#{$end-direction}`? so as to separate it from the streamer name which in some cases could be quite long. 

However it seems to me that the streamer page needs more work(which I am happy to work on) as when I change the language to Arabic the banner appears over the streamers profile picture vs over the
 streamer name in English. Personally I think the live banner looks a lot better over the streamer profile picture as it obscures the streamers username . I have tested `margin-right#{$end-direction}: 10em` with a long streamer name and the streamer language is separated from the streamer name.
<img width="1467" alt="Screenshot 2023-03-11 at 20 29 55" src="https://user-images.githubusercontent.com/20873258/224510238-44a0b6b0-8304-4bbf-8296-0698763d0f37.png">


I also tested it with English. Also as you can see the live banner obscures part of the streamer name which I am happy to fix as well if its an agreed issue that should be fixed. 
<img width="1470" alt="Screenshot 2023-03-11 at 20 31 05" src="https://user-images.githubusercontent.com/20873258/224510278-548e5ae3-006f-4803-ba92-0735d3d6ad79.png">


In both cases the streamer language is separated from the long streamer name so am I correct in saying `margin-right#{$end-direction}` is what is needed to fix the issue or have I completely misunderstood you?

schlawg commented:
I actually think it should be a special case for rtl languages where there's no margin at all.  But I think Benedikt committed the margin-#{$end-direction} change which is close enough.
